src/v0/destinations/salesforce/networkHandler.js

```diff
@@ -1,14 +1,18 @@
 const { proxyRequest, prepareProxyRequest } = require('../../../adapters/network');
 const { processAxiosResponse } = require('../../../adapters/utils/networkUtils');
+const { LEGACY, OAUTH } = require('./config');
 const { salesforceResponseHandler } = require('./utils');
 
 const responseHandler = (destinationResponse, destType) => {
   const message = `Request for destination: ${destType} Processed Successfully`;
 
+  const authorizationFlow = destType === 'salesforce' ? LEGACY : OAUTH;
```

Comment:
Can you explain this logic?

Reply:
when both salesforce and salesforce_oauth will be activated, destType will be either salesforce or salesforce_oauth. for salesforce we will use legacy way of authentication and vice versa

Reply:
can we have a consistent logic to infer LEGACY/OAUTH implementation ?
Here's it's different - https://github.com/rudderlabs/rudder-transformer/blob/9407b951f0d7a81400098492227c3bd3ad8c8644/src/v0/destinations/salesforce/utils.js#L165 ---> relying on destType is fool proof

Reply:
With OAuth - destType would be salesforce_oauth
Without OAuth - destType would be salesforce

`responseHandler` should be sent with `authorizationFlow` arg all the way from controller. We need to figure out which destination type supports which authorization flow.

I think this is the reason the mentioned logic has been added
Please confirm @shrouti1507

Reply:
collectAuthorizationInfo logic also need to be driven via destType - can be different PR

Reply:
added dedicated network handler as discussed